client/src/test/java/org/eclipse/hono/client/device/amqp/EventSenderTest.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.client.device.amqp;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.eclipse.hono.client.AbstractAmqpAdapterClientDownstreamSenderTestBase;
+import org.eclipse.hono.client.device.amqp.internal.AmqpAdapterClientEventSenderImpl;
+import org.junit.jupiter.api.Test;
+
+import io.vertx.core.Future;
+import io.vertx.proton.ProtonDelivery;
+
+/**
+ * Verifies behavior of {@link EventSender}.
+ *
+ */
+public class EventSenderTest extends AbstractAmqpAdapterClientDownstreamSenderTestBase {
+
+    /**
+     * Verifies that the message produced conforms to the expectations of the AMQP adapter.
+     */
+    @Test
+    public void testMessageIsValid() {
+
+        // GIVEN a EventSender instance
+        final EventSender eventSender = AmqpAdapterClientEventSenderImpl
+                .createWithAnonymousLinkAddress(connection, tenantId, s -> {
+                }).result();
+
+        // WHEN sending a message using the API
+        final Future<ProtonDelivery> deliveryFuture = eventSender.send(deviceId, payload, contentType,
```

Comment:
FMPOV this test should at also verify that the EventSender waits for the disposition update from the peer ...